bootstrap.php

```diff
@@ -214,15 +214,15 @@ static function connect() {
         $hosts = explode(',', DBHOST);
         foreach ($hosts as $host) {
             $ferror  = null;
-            if (!db_connect($host, DBUSER, DBPASS, $options)) {
-                $ferror = sprintf('Unable to connect to the database — %s',
-                        db_connect_error());
-            }elseif(!db_select_database(DBNAME)) {
-                $ferror = sprintf('Unknown or invalid database: %s',
-                        DBNAME);
-           }
-           // break if no error
-           if (!$ferror) break;
+            try {
```

Comment:
While this addresses the issue raised, it is also introducing a new bug. Consider the case that `STRICT MySQLi Error Reporting` is disabled (majority of setups)... db_connect will return false - which is not handled in the new logic. Also note that `db_select_database` can also throw an exception IF the database doesn't exist.

I suggest simply wrapping the exising logic with a try / catch - cleaner and handles both cases.

```php

        foreach ($hosts as $host) {
            $ferror  = null;
            try {
                if (!db_connect($host, DBUSER, DBPASS, $options)) {
                    $ferror = sprintf('Unable to connect to the database — %s',
                        db_connect_error());
                } elseif (!db_select_database(DBNAME)) {
                    $ferror = sprintf('Unknown or invalid database: %s',
                        DBNAME);
                }
            } catch (mysqli_sql_exception $e) {
                $ferror = sprintf('Database error — %s', $e->getMessage());
            }
            // break if no error
            if (!$ferror) break;
        }
```